src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -111,10 +117,13 @@ const DomainFragment = ({
     }
     if (from === FROM_FOLLOWED_DOMAIN) {
       const newFollowedDomains = [...newMapFollowedDomain];
-      newFollowedDomains[index].following = !!willFollow;
-      newFollowedDomains[index].user_id_follower = myId;
+      const domainIndex = newFollowedDomains.findIndex(
+        (domain) => domain.domain_name === item.domain_name
+      );
+      newFollowedDomains[domainIndex].following = !!willFollow;
+      // newFollowedDomains[domainIndex].user_id_follower = myId;
 
-      const newDomain = newFollowedDomains[index];
+      const newDomain = newFollowedDomains[domainIndex];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, finding the domain index based on `domain_name` instead of using the array index is a good improvement. This makes the code more robust against changes in the order or number of elements in the `newMapFollowedDomain` array.
```diff
- newFollowedDomains[index].following = !!willFollow;
- newFollowedDomains[index].user_id_follower = myId;
- const newDomain = newFollowedDomains[index];
+ const domainIndex = newFollowedDomains.findIndex(
+   (domain) => domain.domain_name === item.domain_name
+ );
+ newFollowedDomains[domainIndex].following = !!willFollow;
+ // newFollowedDomains[domainIndex].user_id_follower = myId;
+ const newDomain = newFollowedDomains[domainIndex];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->